pkg/gitlab/project_manager.go

```diff
@@ -107,7 +110,8 @@ func (m *ProjectManager) EnsureBranchesAndProtection(project gitlab.Project, dry
 		// Remove protections (if present)
 		if resp, err := m.protectedBranchesClient.UnprotectRepositoryBranches(project.ID, b.Name); err != nil &&
 			(resp == nil || resp.StatusCode != http.StatusNotFound) {
-			return fmt.Errorf("failed to unprotect branch %v before protection: %v", b.Name, err)
+			branchesWithError = append(branchesWithError, fmt.Sprintf("%s: %v", b.Name, err))
```

Comment:
```suggestion
			branchesWithError = append(branchesWithError, fmt.Errorf("%s: %w", b.Name, err)
```

`%w` wraps the error, so you can get it back later by unwrapping the result from `fmt.Errorf`